lib/coverband/utils/railtie.rb

```diff
@@ -11,22 +11,29 @@ def eager_load!
 
   class Railtie < Rails::Railtie
     initializer 'coverband.configure' do |app|
-      app.middleware.use Coverband::BackgroundMiddleware
+      begin
+        app.middleware.use Coverband::BackgroundMiddleware
 
-      if Coverband.configuration.track_views
-        CoverbandViewTracker = Coverband::Collectors::ViewTracker.new
-        Coverband.configuration.view_tracker = CoverbandViewTracker
+        if Coverband.configuration.track_views
+          CoverbandViewTracker = Coverband::Collectors::ViewTracker.new
+          Coverband.configuration.view_tracker = CoverbandViewTracker
 
-        ActiveSupport::Notifications.subscribe(/render_partial.action_view|render_template.action_view/) do |name, start, finish, id, payload|
-          CoverbandViewTracker.track_views(name, start, finish, id, payload) unless name.include?('!')
+          ActiveSupport::Notifications.subscribe(/render_partial.action_view|render_template.action_view/) do |name, start, finish, id, payload|
+            CoverbandViewTracker.track_views(name, start, finish, id, payload) unless name.include?('!')
+          end
         end
+      rescue Redis::CannotConnectError => error
```

Comment:
Would this not be covered by the others rescue block. Think we could reproduce redis being not available with a forked test pointing at a redis instance that does not exist.

Reply:
railtie exceptions can't be covered by the outer block on setup here, because you can't rescue around the `require`, the error doesn't propagate to where you require. 

https://github.com/danmayer/coverband/pull/343/files#diff-a637af9972be5b34783d41387d25e0bfR110

I will see if I can get a test around it

Reply:
Oh gotcha. Makes sense.

Reply:
Just thinking about it some more.. this one may be challenging to test since this code executes as soon as coverband is required.